### PR TITLE
Feature/child object ids

### PR DESCRIPTION
**Pull request by rbuckheit**

### CODE REVIEW

**Comment by nickdhewitt** [comment]
Just merged Drew's changes into develop (we had started his PR earlier in Heracles), so you may want to merge in those updates to make sure nothing breaks.

File: src/main/java/com/kryptnostic/storage/v1/models/request/PendingObjectRequest.java [hunk 1]
```diff
@@ -8,19 +10,33 @@
 
 public class PendingObjectRequest {
     private final String type;
+    private final @Nullable String parentObjectId;
```
Comment (makosblade) [inline src/main/java/com/kryptnostic/storage/v1/models/request/PendingObjectRequest.java]: In the future, I'd like us to move towards using @CheckForNull but this looks fine for now

**Comment by makosblade** [comment]
LGTM

File: src/main/java/com/kryptnostic/storage/v1/models/request/PendingObjectRequest.java (hunk 1 above)
Comment (geekbeast) [inline src/main/java/com/kryptnostic/storage/v1/models/request/PendingObjectRequest.java]: The subtle differences between @CheckForNull and @Nullable seem dangerous. http://stackoverflow.com/questions/12301570/javax-annotation-nullable-vs-checkfornull

File: src/main/java/com/kryptnostic/storage/v1/ObjectIds.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    public static final String CHILD_OBJECT_SEPARATOR = ":";
+
+    public static String getChildObjectId(String objectId, int childIndex) {
+        Validate.isTrue(StringUtils.isNotBlank(objectId));
+        Validate.isTrue(childIndex >= 0);
+        return String.format("%s%s%d", objectId, CHILD_OBJECT_SEPARATOR, childIndex);
+    }
+
+    public static boolean isChildObjectId(String objectId) {
+        return objectId.split(CHILD_OBJECT_SEPARATOR).length > 1;
```
Comment (geekbeast) [inline src/main/java/com/kryptnostic/storage/v1/ObjectIds.java]: Good for now, but long term we should formalize and ID format that is easier to parse like <VERSION>-<TYPE>-<UUID>-<SUB-ID>

**Pull request merged**